File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.java
class Solution 
{
    public List<List<Integer>> levelOrder(Node root) 
    {
         List<List<Integer>> arr = new ArrayList<List<Integer>>();
        if(root == null)
            return arr;
        
        Queue<Node> q = new LinkedList<>();
        q.add(root);
        
        while(q.isEmpty() == false)
        {
            int len = q.size();
            List<Integer> ar = new ArrayList<Integer>();
            while(len > 0)
            {
                Node ptr = q.remove();
                ar.add(ptr.val);
                
                for(int i=0; i<ptr.children.size(); i++)
                    q.add(ptr.children.get(i));
                len--;            
            }          
            arr.add(ar);
        }
        return arr;
    }
}

/*
// Definition for a Node.
class Node 
{
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) 
    {
        val = _val;
    }

    public Node(int _val, List<Node> _children) 
    {
        val = _val;
        children = _children;
    }
};
*/

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.cpp
class Solution 
{
    public:
    vector<vector<int>> levelOrder(Node* root) 
    {
        vector<vector<int>> vec;
        if(root == NULL)
            return vec;
        
        queue<Node *> q;
        q.push(root);
        
        while(q.empty() == false)
        {
            int len = q.size();
            vector<int> v;
            while(len>0)
            {
                Node* ptr = q.front();
                q.pop();
                v.push_back(ptr->val);
            
                for(int i=0; i<ptr->children.size(); i++)
                    q.push(ptr->children[i]);
                len--;
            }
            vec.push_back(v);
        }   
        
        return vec;
    }
};

/*
    class Node 
    {
        public:
        int val;
        vector<Node*> children;

        Node() {}

        Node(int _val) 
        {
            val = _val;
        }

        Node(int _val, vector<Node*> _children) 
        {
            val = _val;
            children = _children;
        }
    };
*/